pandas/tests/io/formats/test_to_xml.py

```diff
@@ -0,0 +1,1401 @@
+        <xsl:value-of select="concat(index, ' ',
+                                     substring($delim, 1, string-length('triangle')
+                                               - string-length(shape) + 1),
+                                     shape,
+                                     substring($delim, 1, string-length(name(degrees))
+                                               - string-length(degrees) + 2),
+                                     degrees,
+                                     substring($delim, 1, string-length(name(sides))
+                                               - string-length(sides) + 2),
+                                     sides)"/>
+         <xsl:text>&#xa;</xsl:text>
+    </xsl:template>
+</xsl:stylesheet>"""
+
+    out_str = geom_df.to_string()
+    out_xml = geom_df.to_xml(na_rep="NaN", stylesheet=xsl)
+
+    assert out_xml == out_str
+
+
+@td.skip_if_no("lxml")
+def test_style_to_json():
+    xsl = """\
+<xsl:stylesheet version="1.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
+    <xsl:output method="text" indent="yes" />
+    <xsl:strip-space elements="*"/>
+
+    <xsl:param name="quot">"</xsl:param>
+
+    <xsl:template match="/data">
+        <xsl:text>{"shape":{</xsl:text>
+        <xsl:apply-templates select="descendant::row/shape"/>
+        <xsl:text>},"degrees":{</xsl:text>
+        <xsl:apply-templates select="descendant::row/degrees"/>
+        <xsl:text>},"sides":{</xsl:text>
+        <xsl:apply-templates select="descendant::row/sides"/>
+        <xsl:text>}}</xsl:text>
+    </xsl:template>
+
+    <xsl:template match="shape|degrees|sides">
+        <xsl:variable name="val">
+            <xsl:if test = ".=''">
+                <xsl:value-of select="'null'"/>
+            </xsl:if>
+            <xsl:if test = "number(text()) = text()">
+                <xsl:value-of select="text()"/>
+            </xsl:if>
+            <xsl:if test = "number(text()) != text()">
+                <xsl:value-of select="concat($quot, text(), $quot)"/>
+            </xsl:if>
+        </xsl:variable>
+        <xsl:value-of select="concat($quot, preceding-sibling::index,
+                                     $quot,':', $val)"/>
+        <xsl:if test="preceding-sibling::index != //row[last()]/index">
+            <xsl:text>,</xsl:text>
+        </xsl:if>
+    </xsl:template>
+</xsl:stylesheet>"""
+
+    out_json = geom_df.to_json()
+    out_xml = geom_df.to_xml(stylesheet=xsl)
+
+    assert out_json == out_xml
+
+
+# COMPRESSION
+
+geom_xml = """\
+<?xml version='1.0' encoding='utf-8'?>
+<data>
+  <row>
+    <index>0</index>
+    <shape>square</shape>
+    <degrees>360</degrees>
+    <sides>4.0</sides>
+  </row>
+  <row>
+    <index>1</index>
+    <shape>circle</shape>
+    <degrees>360</degrees>
+    <sides/>
+  </row>
+  <row>
+    <index>2</index>
+    <shape>triangle</shape>
+    <degrees>180</degrees>
+    <sides>3.0</sides>
+  </row>
+</data>"""
+
+
+def test_bz2_output(parser):
```

Comment:
You can probably parametrize these test for the different compression algorithms.

Not sure: It might be better to commit only the uncompressed `geom_xml` and generate the compressed versions in the test (you can use `get_handle` for that).